hypershift-operator/controllers/hostedcluster/internal/platform/powervs/ccm.go

```diff
@@ -0,0 +1,302 @@
+		return
+	}
+	secretData, secretHasData := secret.Data["ibmcloud_api_key"]
+	if !secretHasData {
+		err = fmt.Errorf("cluster control manager credentials secret %q is missing credentials key", secret.Name)
+		return
+	}
+	apiKeyStr := string(secretData)
+	auth := &ibmcore.IamAuthenticator{
+		ApiKey: apiKeyStr,
+	}
+
+	iamv1, err := iamidentityv1.NewIamIdentityV1(&iamidentityv1.IamIdentityV1Options{
+		Authenticator: auth,
+	})
+	if err != nil {
+		return
+	}
+
+	apiKeyDetailsOpt := iamidentityv1.GetAPIKeysDetailsOptions{IamAPIKey: &apiKeyStr}
+	apiKey, _, err := iamv1.GetAPIKeysDetails(&apiKeyDetailsOpt)
+	if err != nil {
+		return
+	}
+	if apiKey == nil {
+		err = fmt.Errorf("could not retrieve account id")
+		return
+	}
+
+	accountID = *apiKey.AccountID
+	return
+}
+
+func (ccm setupCCM) setupConfigMap() (err error) {
+	accountID, err := ccm.getAccountID()
+	if err != nil {
+		err = fmt.Errorf("error retrieving accountID while setting up config map: %w", err)
+		return
+	}
+
+	configMap := configMap{
+		ConfigFile:               fmt.Sprintf("%s/%s", kubeConfigFileBasePath, kubeConfigKeyInSecret),
+		AccountID:                accountID,
+		ClusterID:                ccm.hcluster.Name,
+		G2workerServiceAccountID: accountID,
+		G2Credentials:            fmt.Sprintf("%s/%s", secretMountPath, "ibmcloud_api_key"),
+		G2ResourceGroupName:      ccm.hcluster.Spec.Platform.PowerVS.ResourceGroup,
+		G2VpcSubnetNames:         ccm.hcluster.Spec.Platform.PowerVS.VPC.Subnet,
+		G2VpcName:                ccm.hcluster.Spec.Platform.PowerVS.VPC.Name,
+		Region:                   ccm.hcluster.Spec.Platform.PowerVS.VPC.Region,
+		PowerVSCloudInstanceID:   ccm.hcluster.Spec.Platform.PowerVS.ServiceInstanceID,
+		PowerVSRegion:            ccm.hcluster.Spec.Platform.PowerVS.Region,
+		PowerVSZone:              ccm.hcluster.Spec.Platform.PowerVS.Zone,
+	}
+
+	templateData := `
+[global]
+version = 1.1.0
+[kubernetes]
+config-file = {{.ConfigFile}}
+[provider]
+cluster-default-provider = g2
+accountID = {{.AccountID}}
+clusterID = {{.ClusterID}}
+g2workerServiceAccountID = {{.G2workerServiceAccountID}}
+g2Credentials = {{.G2Credentials}}
+g2ResourceGroupName = {{.G2ResourceGroupName}}
+g2VpcSubnetNames = {{.G2VpcSubnetNames}}
+g2VpcName = {{.G2VpcName}}
+region = {{.Region}}
+powerVSCloudInstanceID = {{.PowerVSCloudInstanceID}}
+powerVSRegion = {{.PowerVSRegion}}
+powerVSZone = {{.PowerVSZone}}`
+
+	configMapTemplate := template.New("configMap")
+	configMapTemplate.Parse(templateData)
+	configMapData := &bytes.Buffer{}
+	configMapTemplate.Execute(configMapData, configMap)
+
+	configMapObj := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: fmt.Sprintf("%s-%s", ccm.hcluster.Namespace, ccm.hcluster.Name),
+		},
+		Data: map[string]string{
+			configMapName: configMapData.String(),
+		},
+	}
+
+	_, err = ccm.createOrUpdate(ccm.ctx, ccm.c, configMapObj, func() error {
+		return nil
```

Comment:
This is wrong, you need to set the contents of the configmap inside this function and not above in the definition. In the definition, you only set namespace and name